qiskit/assembler/assemble_circuits.py

```diff
@@ -128,7 +128,9 @@ def _assemble_circuit(
             instruction.memory = [clbit_indices[clbit] for clbit in cargs]
             # If the experiment has conditional instructions, assume every
             # measurement result may be needed for a conditional gate.
-            if instruction.name == "measure" and is_conditional_experiment:
+            if (
+                instruction.name == "measure" or instruction.name == "measure_pauli"
```

Comment:
Future developers must be aware that `measure_pauli` exists, and to include checks like the one here when needed. This is tricky.

Reply:
I think this is fine since, in worst scenario, `measure_pauli` will be an unsupported instruction. However,   `MeasurePauli` should not subclass `Measure` for that, in case somebody does a type check.